### DISCUSSION TITLE
Wat verstaan we onder "type" ObservedArea?

### DISCUSSION THREAD

**Discussion by CelineJansen** [kadaster-labs/sensrnet-home#289] [Q&A] [answered]
[EDIT: Geen valide discussie, is inherent aan het Type geometrie (GEOJSON) dat gekozen is]

In de API hebben gebruikers sinds de nieuwe release de optie om een "type" in te geven bij het observatiegebied:

  "observedArea": {
    "type": "string",
    "coordinates": [
      "string"
    ]
  },

Dit type staat niet in de definities, en ook niet in de UI. Dus een beetje gissen wat we hiermee willen.
In de inventarisatielijst heb ik de volgende omschrijving toegevoegd:
"Het type gebied waarover de datastroom informatie geeft. Betreft het een viewrange van de sensor, of juist het gebied waarbinnen een mobiele sensor opereert. Gaat het om gegevens buiten, of ook bijvoorbeeld ook binnen in gebouwen die in het gebied vallen?"

Met dan als voorbeeld:
"Straal waarin de sensor metingen doet. Het maximale gebied is weergegeven, maar het daadwerkelijke gebied kan variëren met de weersomstandigheden."

Ik kan me ook voorstellen dat als je device geen punt is maar meer een gebied (situatie Utrecht met sensoren parkeervlakken die aan een hub zitten als ik het goed heb(?)) dat je dit onder type beschrijft.

Hebben jullie hier nog een mening over? (FAB / pilot deelnemers bijvoorbeeld?)

**Comment by kad-busses** [comment, marked as answer]
Om het type in `observedArea` wat verder toe te lichten: nu bedoelt als een "technisch" type. `observedArea` (als geheel) is nu een GeoJSON Polygon, waarmee het de SensorThings API volgt [(link](http://docs.opengeospatial.org/is/15-078r6/15-078r6.html#:~:text=observedArea,to%2Done%20(optional))). Voor een observatiegebied is het type dan altijd: "Polygon".

In de UI is dat inderdaad niet zichtbaar, maar wordt de optie geboden om een straal te tekenen. Het coördinaat van het device + de straal van de datastroom wordt vervolgens geconverteerd naar de GeoJSON polygoon en als zodanig opgeslagen. De API kent in die zin niet expliciet het concept "straal". Dat wordt gepresenteerd door de applicatie(s) die de API aanspreken.

Dat laatste is interessant, de `observedArea` kan dan gebruikt worden voor het parkeervak zelf. De device zou dan aangemerkt kunnen worden als "toon device locatie niet op de kaart" bijvoorbeeld 😄

**Reply by CelineJansen** [reply]
Ooh wat dom van mij, te snel gesproken. Ik zie nu dat dit gewoon is hoe je GEOJSON eruit komt rollen (bijvoorbeeld: GEOJSON={"type":"Polygon","coordinates":[[[5.48144,51.43652],[5.48144,51.43652],[5.48144,51.43651],[5.48144,51.4365],[5.48143,51.43649],[5.48143,51.43649],[5.48142,51.43648],[5.48142,51.43647],[5.48141,51.43647],[5.4814,51.43646],[5.48139,51.43646],[5.48138,51.43645],[5.48137,51.43645],[5.48136,51.43644],[5.48135,51.43644],[5.48134,51.43644],[5.48132,51.43644],[5.48131,51.43644],[5.4813,51.43644],[5.48129,51.43644],[5.48127,51.43644],[5.48126,51.43644],[5.48125,51.43644],[5.48124,51.43644],[5.48123,51.43645],[5.48122,51.43645],[5.48121,51.43646],[5.4812,51.43646],[5.48119,51.43647],[5.48118,51.43647],[5.48117,51.43648],[5.48117,51.43649],[5.48116,51.43649],[5.48116,51.4365],[5.48116,51.43651],[5.48116,51.43652],[5.48116,51.43653],[5.48116,51.43653],[5.48116,51.43654],[5.48116,51.43655],[5.48116,51.43656],[5.48117,51.43656],[5.48117,51.43657],[5.48118,51.43658],[5.48119,51.43658],[5.4812,51.43659],[5.48121,51.43659],[5.48122,51.4366],[5.48123,51.4366],[5.48124,51.43661],[5.48125,51.43661],[5.48126,51.43661],[5.48127,51.43661],[5.48129,51.43661],[5.4813,51.43661],[5.48131,51.43661],[5.48132,51.43661],[5.48134,51.43661],[5.48135,51.43661],[5.48136,51.43661],[5.48137,51.4366],[5.48138,51.4366],[5.48139,51.43659],[5.4814,51.43659],[5.48141,51.43658],[5.48142,51.43658],[5.48142,51.43657],[5.48143,51.43656],[5.48143,51.43656],[5.48144,51.43655],[5.48144,51.43654],[5.48144,51.43653],[5.48144,51.43652]]]}). Ik dacht dat ik dat type zelf op moest bouwen, maar dat is inherent aan de input formaat. Ik haal "type" weer uit de inventarisatielijst. 

Ik weet niet of jullie daar wat in kunnen wijzigen omdat het automatisch gegenereerd is, maar ipv

  "observedArea": {
    "type": "string",
    "coordinates": [
      "string"
    ]
  },

had ik het meteen door gehad als er gestaan had:

  "observedArea": GEOJSON

(Dan weet je als gebruiker meteen alle eisen, ook aan het lijstje coördinaten.)

**Reply by kad-busses** [reply]
Nee hoor, zeker niet dom! Dit geeft maar mooi aan dat het dus nog niet overal goed en consistent gedocumenteerd is 🙂 Welke inventarisatielijst bedoel je? Lijkt me goed om te kijken of het daar in aangepast kan worden. Inderdaad wel zo handig voor de gebruiker!

**Reply by CelineJansen** [reply]
De lijst die ik centraal gedeeld heb al input voor mijn FME script (daarin had ik dus bij de nieuwe versie onterecht de kolom type voor observedArea toegevoegd...), ik ben nu bezig met de puntjes op de i en zal dan een pull request indienen bij https://github.com/kadaster-labs/sensrnet-home/edit/main/docs/UserManualNL.md met de kloppende documentatie en scripts!

**Reply by kad-busses** [reply]
Ah okay, top! 

Haal je die automatisch gegenereerde types trouwens uit de Swagger UI (die groene pagina op `/api/`)? Daar zie ik die observedArea staan zoals je die omschrijft. Onder "schema" staat er _iets_ meer uitleg, maar dat is dan wel wat ver weggestopt. Valt me daarbij nog wel op dat die niet helemaal correct is: het is een (heel specifiek, maar toch) GeoJSON Polygon en geen ander type. En bij Example Value is "type" is dus altijd "Polygon", en "coordinates" een lijst van lijst van lijst van nummers, niet een ["string"]. Anyway, zal kijken of dat nog aangepast kan worden.

![image](https://user-images.githubusercontent.com/56066664/150377541-a4299e98-78c7-45ad-92f5-ff9f318ece07.png)

**Reply by CelineJansen** [reply]
Jep, ik baseer me inderdaad op die Swagger UI. Dat schema voegt al wel wat toe inderdaad. Maar dit deel "En bij Example Value is "type" dus altijd "Polygon", en "coordinates" een lijst van nummers, niet een ["string"]." bracht voor mij dus inderdaad wat verwarring. Als daar compleet zou staan GeoJSON Polygon (zonder die uitsplitsing van type en coordinates) zou het voor mij duidelijker zijn, dan kan je die uitsplitsing ("type" en "coordinates") en de precieze invulling daarvan om tot een valide GeoJSON te komen vervolgens zelf wel opzoeken via Google of, nog beter, in het schema... Maar goed, smaken verschillen natuurlijk ;).

**Comment by CelineJansen** [comment]
Hij doet het (nu de stroking nog verbeteren ;)):
![image](https://user-images.githubusercontent.com/79267472/150362253-b3981b6a-261f-4e86-9fd1-c47b06149377.png)
